Лаб. работа 1

SemenovKo opened the pull request:
Все получилось, вопросов нет

SemenovKo committed:
Лаб. работа 1

rmushkin commented:
Вопросов, быть может, и нет - но кода я тоже не вижу 😅 
Пожалуйста, загрузите, чтобы я мог проверить, прокомментировать и принять. Кроме того, дайте мне, пожалуйста, права на запись в репозиторий, чтобы я мог принять и закрыть пулл-реквест.

SemenovKo committed:
Попытка загрузить лаб.1

SemenovKo committed:
еще одна попытка загрузки лабы 1

после этапа переноски файлов в окно и нажатия commit changes файлы не появляются

SemenovKo committed:
Лаб. работа 1

SemenovKo committed:
Merge pull request #5 from SemenovKo/lab1

Лаб. работа 1

SemenovKo commented:
Извините, пока плохо разбираюсь в сайте.
Надеюсь, теперь все есть.

rmushkin commented on Лаб. работа номер 1/task2.py:
```diff
@@ -0,0 +1,6 @@
+# TODO Найдите количество книг, которое можно разместить на дискете
+space = 1.44 * 1024 * 1024
+books =  4*25*50*100
+books_in_space = round(space/books)
```
[PEP8](https://peps.python.org/pep-0008/) (указания по стилю написания кода): нужны пробелы вокруг математических операторов

rmushkin replied:
Использование `round` тут опасно: можно округлить, грубо говоря, 99,9 до ста; но если на дискету влезает только 99 книг - то 100 не влезет!
Используйте приведение к целому. Оно просто отбрасывает дробную часть.

rmushkin commented on Лаб. работа номер 1/task1.py:
```diff
@@ -0,0 +1,5 @@
+numbers = [2, -93, -2, 8, None, -44, -1, -85, -14, 90, -22, -90, -100, -8, 38, -92, -45, 67, 53, 25]
+newnumbers = [2, -93, -2, 8, -44, -1, -85, -14, 90, -22, -90, -100, -8, 38, -92, -45, 67, 53, 25]
+# TODO заменить значение пропущенного элемента средним арифметическим
+numbers [4] = sum(newnumbers)/len(numbers)
```
[PEP8](https://peps.python.org/pep-0008/) (указания по стилю написания кода): 
1. если имя переменной состоит из нескольких слов, их нужно разделять нижними подчёркиваниями (не `newnumbers`, а `new_numbers`)
2. Нужны пробелы вокруг математиечких операторов

rmushkin reviewed (changes requested):
Пожалуйста, исправьте ещё пару мелочей

SemenovKo committed:
Update task1.py

SemenovKo committed:
Update task2.py

SemenovKo committed:
Update task1.py

SemenovKo committed:
Update task2.py

SemenovKo commented:
Исправил!

rmushkin commented on Лаб. работа номер 1/task1.py:
```diff
@@ -0,0 +1,5 @@
+numbers = [2, -93, -2, 8, None, -44, -1, -85, -14, 90, -22, -90, -100, -8, 38, -92, -45, 67, 53, 25]
+new_numbers = [2, -93, -2, 8, -44, -1, -85, -14, 90, -22, -90, -100, -8, 38, -92, -45, 67, 53, 25]
+# TODO заменить значение пропущенного элемента средним арифметическим
+numbers [4] = sum(new_numbers) / len(numbers)
```
Заметил ещё, что перед квадратной скобкой пробел не нужен.

rmushkin reviewed (changes requested):
В одном месте затесался беглый пробел.
Кроме того, пожалуйста, удалите одну из папок - пусть на одну лабу будет одна.

SemenovKo committed:
Delete Лаб. работа 1 directory

SemenovKo committed:
Update task1.py
